apps/opik-backend/src/main/java/com/comet/opik/api/sorting/SortingFactory.java

```diff
@@ -0,0 +1,51 @@
+package com.comet.opik.api.sorting;
+
+import com.comet.opik.utils.JsonUtils;
+import com.fasterxml.jackson.core.type.TypeReference;
+import jakarta.ws.rs.BadRequestException;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class SortingFactory {
+    public static final String ERR_INVALID_SORTING_PARAM_TEMPLATE = "Invalid sorting query parameter '%s'";
+    public static final String ERR_ILLEGAL_SORTING_FIELDS_TEMPLATE = "Invalid sorting fields '%s'";
+    public List<SortingField> newSorting(String queryParam) {
+        List<SortingField> sorting = new ArrayList<>();
+
+        if (StringUtils.isBlank(queryParam)) {
+            return sorting;
+        }
+
+        try {
+            sorting = JsonUtils.readValue(queryParam, new TypeReference<>() {
```

Comment:
I'm not sure, but `TypeReference` is a special class that helps Jackson perform the runtime serialization.  Using it like this may lead issues.

Reply:
I use it to read the `List`. The alternative is extending `JsonUtils` with something like:
```java
public <T> T readCollectionValue(@NonNull String content, @NonNull Class<? extends Collection> collectionClass,
                                     @NonNull Class<?> valueClass) {
    try {
        return MAPPER.readValue(content, JsonUtils.MAPPER.getTypeFactory().constructCollectionType(
                collectionClass, valueClass));
    } catch (JsonProcessingException exception) {
        throw new UncheckedIOException(exception);
    }
}
```
Will do that.